view/src/main/java/org/xcolab/view/auth/endpoints/UserTrackingController.java

```diff
@@ -60,15 +70,41 @@ private String getHeadersAsString(HttpServletRequest request) {
 
         while (headerNames.hasMoreElements()) {
             String headerName = headerNames.nextElement();
-            Enumeration<String> headers = request.getHeaders(headerName);
+            if (StringUtils.equalsAnyIgnoreCase(headerName, IGNORED_HEADERS)) {
+                continue;
+            }
 
+            Enumeration<String> headers = request.getHeaders(headerName);
             while (headers.hasMoreElements()) {
-                headerStringBuilder.append(headers.nextElement()).append("\n");
+                String headerValue = headers.nextElement();
+                if (headerName.equalsIgnoreCase(HttpHeaders.COOKIE)) {
+                    headerValue = removeCookiesByName(headerValue, IGNORED_COOKIES);
+                }
+                headerStringBuilder.append(headerName).append(": ");
+                headerStringBuilder.append(headerValue).append("\n");
             }
         }
         return headerStringBuilder.toString();
     }
 
+    private String removeCookiesByName(String cookieHeader, String... names) {
+        return Arrays.stream(cookieHeader.split("\\s*;\\s*"))
+                .filter(c -> !matchesAnyCookieName(c, names))
+                .collect(Collectors.joining("; "));
+    }
+
+    private boolean matchesAnyCookieName(String cookieString, String... cookieNames) {
+        //noinspection SimplifiableIfStatement
+        if (StringUtils.isEmpty(cookieString) || ArrayUtils.isEmpty(cookieNames)) {
```

Comment:
Why this separate `if` statement and not just `return isEmpty(cookieString) || isEmpty(cookieNames) || Arrays.stream(...`?

Reply:
I like the separation between an exceptional case, the input being empty, and the normal processing. Using separate statements makes it more clear that we are checking for two different things. I just pushed a change that represents this separation as a local variable and uses a single conditional in the return. What do you think about that version?